tests/README.md

```diff
@@ -1,24 +1,130 @@
+and is the simplest way to run the tests locally in a development environment.
 
+To run the test locally:
+- Configure the desired cluster setup (provider, number of nodes and nics).
+
+   > **Note:** Some tests may run only on specific configuration and hardware.
+
+   > **Note:** For detailed instructions on how to work with a specific provider,
+        please explore kubevirtci providers, e.g.
+  [k8s-1.18](https://github.com/kubevirt/kubevirtci/tree/master/cluster-up/cluster/k8s-1.18)
+  
+  The following configuration is of a k8s-1.18 provider with two nodes, each
+  having two additional interfaces (3 in total).
 ```
-cd tests # from the git repo root folder
-go test -kubeconfig=path/to/my/config -config=default-config.json
+export KUBEVIRT_PROVIDER=k8s-1.18
+export KUBEVIRT_NUM_NODES=2
+export KUBEVIRT_NUM_SECONDARY_NICS=2
 ```
 
-## Run them on one of the core KubeVirt providers
+- Raise a cluster based on the configuration.
+  At this stage, it will not contain kubevirt.
+  ```
+  make cluster-up
+  ```
+- Build KubeVirt from sources.
+  ```
+  make
+  ```
+- Deploy KubeVirt onto the cluster.
+  ```
+  make cluster-sync
+  ```
+- Run the tests.
+  ```
+  make functest
+  ```
+- Take down and delete the cluster.
+  ```
+  make cluster-down
+  ```
+
+On a regular workflow of development or debugging, rebuilding KubeVirt is
+needed only when the production code is changed.
+In case only test code changes, `make functest` will rebuild the test code
+implicitly.
+
+## Existing (external) Kubevirt Cluster
+
+The tests may be executed against an already deployed (and active) Kubevirt
+cluster.
+
+The environment needs to be adjusted to allow the tests to operate against
+such a cluster.
+The following high level steps are required to accomplish this:
+- Clone kubevirt repository on the machine that has access to the cluster.
+- Identify the deployed kubevirt version.
+- From the git repo, checkout to the commit that correlates to the running
+kubevirt cluster.
+- Run the tests by executing the
+  [./tests/run-tests-on-external-cluster.sh](./run-tests-on-external-cluster.sh)
+  script.
+
+> **Note:** In case the external cluster uses dynamic storage provisioner
+such as hostpath, then there is a need to provide a specific custom storage
+configuration through a configuration file.
+This is currently not covered by this document and the script helper.
+
+### Identify the commit on which the deployed kubevirt cluster is based on
 
-There is a make target to run this with the config
-taken from hack/config.sh:
+In order to identify on which commit the current kubevirt cluster is based on,
+there is a need to query a kubevirt infra pod and then inspect its base
+container.
 
+- Lookup for the `virt-operator` pod:
+  `kubectl get pods --all-namespaces | grep "virt-operator"`
+- Use the full pod name and its namespace to inspect the container it uses:
+  `kubectl describe <virt-operator full name> -n <virt-operator namespace>`
```

Comment:
all this section might worth a little script ? or some lines you can put in the readme that would make it for you?
some docs already use this approach, and it makes life simpler

Reply:
There is no script for this.
If this will come from a partner or a developer will be interested to do it, we can add it (the later can send a PR).